frontend/catalyst/compiler.py

```diff
@@ -242,7 +242,8 @@ def lower_mhlo_to_linalg(filename):
     Returns:
         a path to the output file
     """
-    assert filename[-5:] == ".mlir", "input is not an mlir file"
+    if filename.split(os.extsep)[-1] != "mlir":
```

Comment:
Would this not fail if `os.extsep` happens to be different from `.`, given that we hardcode this character when we generate these files?

Reply:
A good point indeed. I think we don't want to have the perfect OS compatibility here, so I think I'll rollback to the original version.

Reply:
Reverted